File: src/main/java/com/hexaware/cricket/service/PlayerService.java
package com.hexaware.cricket.service;
import com.hexaware.cricket.dto.PlayerRequestDTO;
import com.hexaware.cricket.dto.PlayerResponseDTO;
import java.util.List;

public interface PlayerService {
	List<PlayerResponseDTO> getAllPlayers();
	PlayerResponseDTO getPlayerById(Long playerId);
	PlayerResponseDTO createPlayer(PlayerRequestDTO requestDTO);
	PlayerResponseDTO updatePlayer(Long playerId,PlayerRequestDTO requestDTO);
	void deletePlayer(Long playerId);
	List<PlayerResponseDTO> getPlayersByStateName(String stateName);
}

File: frontend/src/services/PlayerService.js
import axios from 'axios';

const BASE_URL = 'http://localhost:8080/api/players';

const PlayerService = {
    getAllPlayers: () => {
        return axios.get(BASE_URL);
    },
    getPlayerById: (playerId) => {
        return axios.get(`${BASE_URL}/${playerId}`);
    },
    createPlayer: (playerData) => {
        return axios.post(BASE_URL, playerData);
    },
    updatePlayer: (playerId, playerData) => {
        return axios.put(`${BASE_URL}/${playerId}`, playerData);
    },
    deletePlayer: (playerId) => {
        return axios.delete(`${BASE_URL}/${playerId}`);
    },
    getPlayersByState: (stateName) => {
        return axios.get(`${BASE_URL}/state/${stateName}`);
    }
};

export default PlayerService;